reporter.go

```diff
@@ -0,0 +1,128 @@
+package check
+
+import (
+	"fmt"
+	"io"
+	"sync"
+)
+
+type reporter interface {
```

Comment:
failure, error, success, skip and expected failure are what define a unit test reporter, I think the interface must have them. 
started and missed are things that I don't like, but they come from the gocheck implementation. We could just remove them, but then we are changing the gocheck behaviour.
stream and write I think that should be internal to the gocheck default reporter, but they are used in gocheck.go, so again that would require more refactors than what this PR intents to do.

Do you have a suggestion about how to split the interface?